flang/lib/Optimizer/CodeGen/CodeGen.cpp

```diff
@@ -369,7 +403,7 @@ class FIROpConversion : public mlir::ConvertOpToLLVMPattern<FromOp> {
   }
 
   // Generate an alloca of size 1 for an object of type \p llvmObjectTy.
-  mlir::LLVM::AllocaOp
+  mlir::Value
   genAllocaWithType(mlir::Location loc, mlir::Type llvmObjectTy,
```

Comment:
Given that this util now generates either an `alloca` or a `cast`, can you update the name to reflect this?